flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalogBase.java

```diff
@@ -197,4 +237,196 @@ protected void alterHiveDatabase(String name, Database newHiveDatabase, boolean
+				true,
+				ignoreIfNotExists);
+		} catch (NoSuchObjectException e) {
+			if (!ignoreIfNotExists) {
+				throw new TableNotExistException(catalogName, tablePath);
+			}
+		} catch (TException e) {
+			throw new CatalogException(
+				String.format("Failed to drop table %s", tablePath.getFullName()), e);
+		}
+	}
+
+	@Override
+	public List<String> listTables(String databaseName)
+			throws DatabaseNotExistException, CatalogException {
+		try {
+			return client.getAllTables(databaseName);
+		} catch (UnknownDBException e) {
+			throw new DatabaseNotExistException(catalogName, databaseName);
+		} catch (TException e) {
+			throw new CatalogException(
+				String.format("Failed to list tables in database %s", databaseName), e);
+		}
+	}
+
+	@Override
+	public List<String> listViews(String databaseName) throws DatabaseNotExistException, CatalogException {
+		try {
+			return client.getTables(
+				databaseName,
+				null, // table pattern
+				TableType.VIRTUAL_VIEW);
+		} catch (UnknownDBException e) {
+			throw new DatabaseNotExistException(catalogName, databaseName);
+		} catch (TException e) {
+			throw new CatalogException(
+				String.format("Failed to list views in database %s", databaseName), e);
+		}
+	}
+
+	@Override
+	public boolean tableExists(ObjectPath tablePath) throws CatalogException {
+		try {
+			return client.tableExists(tablePath.getDatabaseName(), tablePath.getObjectName());
+		} catch (UnknownDBException e) {
+			return false;
+		} catch (TException e) {
+			throw new CatalogException(
+				String.format("Failed to check whether table %s exists or not.", tablePath.getFullName()), e);
+		}
+	}
+
+	private Table getHiveTable(ObjectPath tablePath) throws TableNotExistException {
+		try {
+			return client.getTable(tablePath.getDatabaseName(), tablePath.getObjectName());
+		} catch (NoSuchObjectException e) {
+			throw new TableNotExistException(catalogName, tablePath);
+		} catch (TException e) {
+			throw new CatalogException(
+				String.format("Failed to get table %s from Hive metastore", tablePath.getFullName()), e);
+		}
+	}
+
+	/**
+	 * Create a Flink's TableSchema from Hive table's columns and partition keys.
+	 */
+	protected TableSchema createTableSchema(List<FieldSchema> cols, List<FieldSchema> partitionKeys) {
+		List<FieldSchema> allCols = new ArrayList<>(cols);
+		allCols.addAll(partitionKeys);
+
+		String[] colNames = new String[allCols.size()];
+		TypeInformation[] colTypes = new TypeInformation[allCols.size()];
+
+		for (int i = 0; i < allCols.size(); i++) {
+			FieldSchema fs = allCols.get(i);
+
+			colNames[i] = fs.getName();
+			colTypes[i] = HiveTypeUtil.toFlinkType(TypeInfoUtils.getTypeInfoFromTypeString(fs.getType()));
+		}
+
+		return new TableSchema(colNames, colTypes);
+	}
+
+	/**
+	 * Create Hive columns from Flink TableSchema.
+	 */
+	protected List<FieldSchema> createHiveColumns(TableSchema schema) {
```

Comment:
static?

Reply:
I think this is fine as it's used and only used by its two subclasses, not any other classes. Later on we may be able to abstract column and partition creation from subclasses to base class, and this would save us from debating whether it's should be base class or a util class back and forth.

Reply:
It probably doesn't matter much if one develops both parent class and child classes. Image the base class is defined by one different person, who probably doesn't have much idea what the child class needs. Putting the two utility methods here suggests the parent class needs to envision what's needed by the child classes, which doesn't meet the criteria of separation of concerns. I think moving the two methods to static and out of the parent class would make the code cleaner.